pass.go

```diff
@@ -0,0 +1,1054 @@
+		} else if netErr, okNetErr := passErr.error.(net.Error); okNetErr {
+			if opErr, okOpErr := netErr.(*net.OpError); okOpErr {
+				if dnsErr, okDNSErr := opErr.Err.(*net.DNSError); okDNSErr {
+					log.Printf("DNS Lookup error, temporary: %t, timeout: %t: %v", dnsErr.IsTemporary, dnsErr.IsTimeout, dnsErr)
+					return http.StatusInternalServerError, p.sv.templates.Errors500, ""
+				} else if osSyscallErr, okOsSyscallErr := opErr.Err.(*os.SyscallError); okOsSyscallErr {
+					log.Printf("Syscall %s resulted in code %d during %s in: %v", osSyscallErr.Syscall, osSyscallErr.Err, opErr.Op, opErr)
+					return http.StatusInternalServerError, p.sv.templates.Errors500, ""
+				}
+
+				log.Printf("Generic *net.OpError: %v", opErr)
+				return http.StatusInternalServerError, p.sv.templates.Errors500, ""
+			}
+
+			log.Printf("net.Error: %v", netErr)
+			return http.StatusInternalServerError, p.sv.templates.Errors500, ""
+		} else if osPathError, okOsPathError := passErr.error.(*os.PathError); okOsPathError {
+			log.Printf("*os.PathError: %v", osPathError)
+			return http.StatusInternalServerError, p.sv.templates.Errors500, ""
+		} else if passErr.Error() == "UUID not found" {
+			return http.StatusNotFound, p.sv.templates.ErrorsPwNotFound, passErr.Error()
+		}
+
+		log.Printf("passErr: %v", passErr)
+		return http.StatusInternalServerError, p.sv.templates.Errors500, ""
+	}
+
+	if ivErr, okIvErr := err.(passErrorInputValidation); okIvErr {
+		if ivErr.tooLong || ivErr.tooShort {
+			s := "long"
+
+			if ivErr.tooShort {
+				s = "short"
+			}
+
+			return http.StatusBadRequest, p.sv.templates.Errors400, fmt.Sprintf("Your password is too %s", s)
+		}
+	}
+
+	if httpErr, okHTTPErr := err.(httpError); okHTTPErr {
+		if httpErr.code == http.StatusBadRequest {
+			return httpErr.code, p.sv.templates.Errors400, httpErr.Error()
+		} else if httpErr.code == http.StatusNotFound {
+			return httpErr.code, p.sv.templates.Errors404, httpErr.Error()
+		} else if httpErr.code == http.StatusMethodNotAllowed {
+			return httpErr.code, p.sv.templates.Errors405, httpErr.Error()
+		}
+
+		log.Printf("httpErr: %v", err)
+		return httpErr.code, p.sv.templates.Errors500, httpErr.Error()
+	}
+
+	log.Printf("UNHANDLED Error: %v", err)
+	return http.StatusInternalServerError, p.sv.templates.Errors500, ""
+}
+
+// NewHTTPError returns an instance of httpError which is required for
+// RenderErrorPage for proper display and error logging
+func NewHTTPError(code int, parentError error) error {
+	return httpError{code: code, error: parentError}
+}
+
+// ResolveTemplatePath returns the full path to the requested template file
+func (p *Pass) ResolveTemplatePath(templateSubPath string) (resolvedTemplatePath string) {
+	resolvedTemplatePath = filepath.Join(p.sv.cfg.TemplatePath, templateSubPath)
+
+	if _, err := os.Stat(resolvedTemplatePath); err != nil {
+		if os.IsNotExist(err) {
+			log.Fatalf("the template file %s was requested but it does not exist", resolvedTemplatePath)
```

Comment:
Why not just log out the original `err`? We don't need to add our own commentary.

Reply:
Fixed in next push